src/xamarin/sk_manageddrawable.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2015 Xamarin Inc.
+ * Copyright 2017 Microsoft Corporation. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include "SkManagedDrawable.h"
+
+#include "sk_manageddrawable.h"
+#include "sk_picture.h"
+#include "sk_types_priv.h"
+
+
+static sk_manageddrawable_draw_delegate               gDraw;
+static sk_manageddrawable_getBounds_delegate          gGetBounds;
+static sk_manageddrawable_newPictureSnapshot_delegate gNewPictureSnapshot;
+
+static inline SkManagedDrawable* AsManagedDrawable(sk_manageddrawable_t* cdrawable) 
+{
+    return reinterpret_cast<SkManagedDrawable*>(cdrawable);
+}
+
+static inline sk_manageddrawable_t* ToManagedDrawable(SkManagedDrawable* drawable) 
+{
+    return reinterpret_cast<sk_manageddrawable_t*>(drawable);
+}
+
+static inline const sk_manageddrawable_t* ToManagedDrawable(const SkManagedDrawable* drawable) 
+{
+    return reinterpret_cast<const sk_manageddrawable_t*>(drawable);
+}
+
+void dDraw(SkManagedDrawable* managedDrawable, SkCanvas* canvas)
+{
+    return gDraw(ToManagedDrawable(managedDrawable), ToCanvas(canvas));
```

Comment:
we can't return `void`.

```suggestion
    gDraw(ToManagedDrawable(managedDrawable), ToCanvas(canvas));
```

Reply:
Done